### PR TITLE
Feature/bulk schedule

### LINKED ISSUE #326: Post schedule
Currently the post endpoint for student_on_tour (/student_on_tour/) receives a singular object in the body, this should be changed to an array of objects so a new schedule can be posted as whole entity.

### PR DESCRIPTION
**Pull request by sevrijss**
This PR adds bulk methods to the StudentOnTour endpoint, so that the frontend doesn't need to make a million request when assigning students to tours.

Reviewers:
* Jonathan: review code and possibly test
* Emma: read the docs and confirm this is the functionality you wanted, if not leave a request for changes and explain what you might need extra / different.

closes #326

### CODE REVIEW

**Review by emneirin — APPROVED** [review]
Yes thank you this is what I wanted from functionality

File: backend/student_on_tour/views.py [hunk 1]
```diff
@@ -31,6 +30,151 @@ def post(self, request):
...
+            [
+                {Tour:x, student:x, date: x},
+                {Tour:x2, student:x, date: x2},
+                // more of this
+            ]
+        }
+        """
+        list_done = []
+        for d in data["data"]:
+            print(d)
```
Comment (jonathancasters) [inline backend/student_on_tour/views.py:72]: Forgot to remove print statement?

File: backend/student_on_tour/views.py [hunk 2]
```diff
@@ -31,6 +30,151 @@ def post(self, request):
...
+            student_on_tour_instance = StudentOnTour()
+
+            set_keys_of_instance(student_on_tour_instance, d, TRANSLATE)
+
+            if r := try_full_clean_and_save(student_on_tour_instance):
+                for elem in list_done:
+                    elem.delete()
+                return r
+            list_done.append(student_on_tour_instance)
```
Comment (jonathancasters) [inline backend/student_on_tour/views.py:81]: Why are you saving the saving the student_on_tour_instances in this list_done?

File: backend/student_on_tour/views.py (hunk 1 above)
Comment (sevrijss) [reply backend/student_on_tour/views.py:72]: You're right :)

File: backend/student_on_tour/views.py (hunk 2 above)
Comment (sevrijss) [reply backend/student_on_tour/views.py:81]: It's a history of the previous saves, so if 1 fails along the way, I can remove all the previous ones as well.

**Review by jonathancasters — APPROVED** [review]

**Comment by sevrijss** [comment]
Merged - testing succeeds local

**Pull request merged**